gcloud-java-storage/src/main/java/com/google/gcloud/storage/BlobInfo.java

```diff
@@ -259,98 +296,177 @@ private BlobInfo(Builder builder) {
     return blobId;
   }
 
+  /**
+   * Returns the name of the containing bucket.
+   */
   public String bucket() {
     return blobId().bucket();
   }
 
+  /**
+   * Returns the blob's id.
+   */
   public String id() {
     return id;
   }
 
+  /**
+   * Returns the blob's name.
+   */
   public String name() {
     return blobId().name();
   }
 
+  /**
+   * Returns the blob's data cache control.
+   */
   public String cacheControl() {
     return Data.isNull(cacheControl) ? null : cacheControl;
   }
 
+  /**
+   * Returns the blob's access control configuration.
+   *
+   * @see <a href="https://cloud.google.com/storage/docs/access-control#About-Access-Control-Lists">
+   *     About Access Control Lists</a>
+   */
   public List<Acl> acl() {
     return acl;
   }
 
+  /**
+   * Returns the blob's owner.
+   */
   public Acl.Entity owner() {
     return owner;
   }
 
+  /**
+   * Returns the blob's data size in bytes.
+   */
   public Long size() {
     return size;
   }
 
+  /**
+   * Returns the blob's data content type.
+   */
   public String contentType() {
     return Data.isNull(contentType) ? null : contentType;
   }
 
+  /**
+   * Returns the blob's data content encoding.
+   */
   public String contentEncoding() {
     return Data.isNull(contentEncoding) ? null : contentEncoding;
   }
 
+  /**
+   * Returns the blob's data content disposition.
+   */
   public String contentDisposition() {
     return Data.isNull(contentDisposition) ? null : contentDisposition;
   }
 
+  /**
+   * Returns the blob's data content language.
+   */
   public String contentLanguage() {
     return Data.isNull(contentLanguage) ? null : contentLanguage;
   }
 
+  /**
+   * Returns the number of components that make up this object. Components are accumulated through
+   * the {@link Storage#compose(Storage.ComposeRequest)} operation.
+   *
+   * @see <a href="https://cloud.google.com/storage/docs/composite-objects#_Count">Component Count
+   *     Property</a>
+   */
   public Integer componentCount() {
     return componentCount;
   }
 
+  /**
+   * Returns blob resource's entity tag.
+   */
   public String etag() {
     return etag;
   }
 
+  /**
+   * Returns the URI of this blob as a string.
+   */
   public String selfLink() {
     return selfLink;
   }
 
+  /**
+   * Returns the MD5 hash of blob's data encoded in base64.
+   */
   public String md5() {
     return Data.isNull(md5) ? null : md5;
   }
 
+  /**
+   * Returns the CRC32C checksum of blob's data encoded in base64 in big-endian order.
+   */
   public String crc32c() {
     return Data.isNull(crc32c) ? null : crc32c;
   }
 
+  /**
+   * Returns the blob's media download link.
+   */
   public String mediaLink() {
     return mediaLink;
   }
 
+  /**
+   * Returns blob's metadata.
+   */
   public Map<String, String> metadata() {
     return metadata == null || Data.isNull(metadata) ? null : Collections.unmodifiableMap(metadata);
   }
 
+  /**
+   * Returns blob's data generation.
```

Comment:
Did you consider copy/paste from here https://cloud.google.com/storage/docs/json_api/v1/objects#resource.
I think some of the descriptions there are nice.